src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -358,4 +371,24 @@ private JsonObject toJson(EncryptionKey key) {
         jo.put("expires", key.getExpires().toEpochMilli());
         return jo;
     }
-}
+
+    private boolean isWithinCutOffTime(EncryptionKey key, Instant now, boolean duringRotation) {
+        if (!(filterKeyOverCutOffTime && duringRotation)) {
+            return true;
+        }
+        int siteId = key.getSiteId();
+        Duration cutoffTime;
+        switch (siteId) {
+            case -1:
```

Comment:
sorry forgot to mention should use Const.Data.MasterKeySiteId /Const.Data.RefreshKeySiteId here instead of hardcoded numbers